### PR TITLE
chore: upgrade to Truffle 5 and solidity-coverage 0.7

### PR DESCRIPTION
**Pull request by bingen**

### CODE REVIEW

File: apps/token-wrapper/truffle-config.js [hunk 1]
```diff
@@ -1 +1,21 @@
-module.exports = require('@aragon/truffle-config-v4')
+const truffleConfig = require('@aragon/truffle-config-v4')
```
Comment (sohkai) [inline apps/token-wrapper/truffle-config.js]: We should be able to just use `@aragon/truffle-config-v5` :)

File: shared/contract-utils/test/helpers/assertThrow.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+const THROW_ERROR_PREFIX = 'VM Exception while processing transaction: revert'
```
Comment (sohkai) [inline shared/contract-utils/test/helpers/assertThrow.js]: Are these changes to `assertThrow()` needed?

As far as I can see, the only change is to remove [this line](https://github.com/aragon/aragon-apps/blob/master/shared/test-helpers/assertThrow.js#L25), which by itself shouldn't cause any problems (unless `error.message` is now non-existent?).

**Review by sohkai — COMMENTED** [review]
Have a few comments, but looks good otherwise.

File: shared/contract-utils/test/helpers/assertThrow.js (hunk 2 above)
Comment (bingen) [reply shared/contract-utils/test/helpers/assertThrow.js]: Yes, it's needed: the structure of `error.message` is now different, so it never matches the string. Besides, having now `error.reason` with what we are looking for directly is way more elegant and trustworthy.

Comment (sohkai) [reply shared/contract-utils/test/helpers/assertThrow.js]: Sounds like this is something we should handle generically across truffle v4 and v5 in contract-helpers then.

**Review by sohkai — APPROVED** [review]
🙌 Finally getting this through the finish line... thanks for your earlier work @bingen!

I've just updated it a bit and merged with master :).

**Pull request merged**